src/lib/types.ts

```diff
@@ -0,0 +1,34 @@
+declare global {
+  // for some reason, the Window constructor does not exist on the window object :/
+  interface Window {
+    Window: () => Window;
+  }
+}
+
+type UnsubscribeMethod = (event: string, fn: SubscribeHandler) => boolean;
+type SubscribeMethod = (event: string, fn: SubscribeHandler) => boolean;
+type PublishMethod = (
+  event: string,
+  data?: SubscriberArg | SubscribeHandler,
+  reply?: SubscribeHandler
+) => boolean;
+type ReplyFunction = (...args: unknown[]) => void;
+
+export type FramebusEvent = {
+  data: string;
+};
+
+export type FramebusPayload = {
+  data?: string;
+  event: string;
+  origin: string;
+  reply?: string | ReplyFunction;
+  eventData?: SubscriberArg;
+};
+export type SubscriberArg = Record<string, unknown>;
+export type SubscribeHandler = (
```

Comment:
Not sure what the right way to set this is. With this integration:

```ts
import * as bus from "framebus";

bus.on("foo", (data: Record<string, string>) => {
  console.log('foo called', data.foo);
});

const msg = {
  foo: 'foo'
}

bus.emit('foo', msg);
```

I get:

```
ERROR in /home/admin/bt/ts-demo/src/index.ts
./src/index.ts
[tsl] ERROR in /home/admin/bt/ts-demo/src/index.ts(3,15)
      TS2345: Argument of type '(data: Record<string, string>) => void' is not assignable to parameter of type 'SubscribeHandler'.
  Types of parameters 'data' and 'data' are incompatible.
    Type 'SubscribeHandler | Record<string, unknown> | undefined' is not assignable to type 'Record<string, string>'.
      Type 'undefined' is not assignable to type 'Record<string, string>'.

```